controllers/datadogagent/feature/apm/feature.go

```diff
@@ -97,6 +126,21 @@ func (f *apmFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.Requ
 				},
 			},
 		}
+		if apm.SingleStepInstrumentation != nil &&
+			(dda.Spec.Features.AdmissionController != nil && apiutils.BoolValue(dda.Spec.Features.AdmissionController.Enabled)) {
+			// add debug log in case Admission controller is disabled (it's a required feature).
+			f.singleStepInstrumentation = &instrumentationConfig{}
+			f.singleStepInstrumentation.enabled = apiutils.BoolValue(apm.SingleStepInstrumentation.Enabled)
```

Comment:
a few lines above we set this if `len(apmConf.SingleStepInstrumentation.EnabledNamespaces) > 0`, maybe we should do the same